### PR TITLE
Updated indicators

### PR DESCRIPTION
**Pull request by ludwiczakpawel**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Documentation Update

## Description

This PR slightly updates our indicator (little colored badges with number or short label) as well as introduces proper Storybook documentation and controls.

It slightly updates the code:
- I renamed `crayons-indicator` to be `c-indicator`
- I added few variants (styles) and extra configuration (`relaxed` prop) needed for other projects.

This is something we wanted to update for Admin's Member Details view that is currently being in progress so I thought it's a good moment to update this.

We don't have too many instances of this component used across the codebase so I just went ahead and updated these everywhere, so `crayons-indicator` should no longer be there.

Keep in mind most of the changed files in this PR is simply coming from renaming `crayons-indicator` to `c-indicator` 🤷 

## Documents

[Figma source file](https://www.figma.com/file/gHpwugeu9jFGW4d9u3ReuR/Crayons?node-id=391%3A234) - keep in mind we have different colors in Figma because our Figma document already uses new colors palette which is not yet implemented in the codebase - it's currently being reviewed in [separate PR](https://github.com/forem/forem/pull/16107). I will of course update everything once we ship that PR.

## QA Instructions, Screenshots, Recordings

**Storybook:** Run `yarn storybook` and go to Indicators section which should now contain correct controls and documentation.
**Product:** Just browse around. Even though all indicators have been "updated" - changes aren't really visually significant or meaningful to raise anyone's concerns or to accidentally break anything.

### UI accessibility concerns?

I've checked the contrast ratio for each variant (style) for both themes (light & dark).

## Added/updated tests?

- [x] No, should I?

## [Forem core team only] How will this change be communicated?

Forem.team Crayons changelog post.

### CODE REVIEW

**Review by aitchiss — COMMENTED** [review]
Looks good - I just have a couple of small questions

File: app/javascript/crayons/Indicators/__stories__/Indicators.mdx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- Success - green.
+- Warninig - yellow.
+- Danger - red.
+
+Even though they are named specifically, they don't really come with any function in mind - main difference is basically a color.
+
+## Padding
+
+By default, Indicator has very tight padding around label - `4px`. It's ok for short labels - like a number.
+But for longer labels it may look better with some extra spacing. And there comes `relaxed` prop which will increase the padding to `8px`.
```
[the remark this answers is not in the document]
Comment (ludwiczakpawel) [reply app/javascript/crayons/Indicators/__stories__/Indicators.mdx:25]: Yea, probably :D  I'll update it.

Also, you're no fun!

[the remark this answers is not in the document]
Comment (aitchiss) [reply app/javascript/crayons/Indicators/__stories__/Indicators.mdx:25]: > Also, you're no fun!

NOPE

File: app/assets/stylesheets/components/indicators.scss [hunk 2]
```diff
@@ -1,60 +1,39 @@
...
-    critical,
-    var(--indicator-critical-bg),
-    var(--indicator-critical-color)
-  );
+  &--warning {
+    --bg: var(--badge-warning-bg);
+    --color: var(--badge-warning-color);
+  }
+
+  &--danger {
```
[the remark this answers is not in the document]
Comment (ludwiczakpawel) [reply app/assets/stylesheets/components/indicators.scss:24]: That's weird, I did check the contrast ratio in Storybook, 100%!

[the remark this answers is not in the document]
Comment (ludwiczakpawel) [reply app/assets/stylesheets/components/indicators.scss:24]: Oh, interesting: these cards on that particular view have `opacity-75` that reduces opacity of everything FOR SOME REASON. So the contrast ratio of this indicator is actually correct - it's the opacity of parent element that ruins it...

File: app/javascript/crayons/Indicators/Indicator.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import classNames from 'classnames/bind';
+
+export const Indicator = (props) => {
+  const {
+    children,
+    variant = 'default',
+    relaxed,
+    className,
+    ...otherProps
+  } = props;
```
[the remark this answers is not in the document]
Comment (ludwiczakpawel) [reply app/javascript/crayons/Indicators/Indicator.jsx:12]: fixing

File: app/assets/stylesheets/components/indicators.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (ludwiczakpawel) [reply app/assets/stylesheets/components/indicators.scss:24]: I'll sneak in a fix for this as i don't think having that opacity-75 was a right call anyway.

**Comment by ludwiczakpawel** [comment]
@aitchiss thank you, Senior Front End Developer! Everything fixed :D

File: app/assets/stylesheets/components/indicators.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (aitchiss) [reply app/assets/stylesheets/components/indicators.scss:24]: oh weird! in storybook it has a ratio of 5.38, but then in axe dev tools it's 3.49 🤔 

I think it's axe devtools having a headache - running the colours through another checker agrees with storybook. So... nothing to see here 😄

**Review by aitchiss — APPROVED** [review]

File: app/assets/stylesheets/components/indicators.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (aitchiss) [reply app/assets/stylesheets/components/indicators.scss:24]: haha our messages crossed!

VERY interesting - I'm glad it makes sense now 😄

**Review by nickytonline — APPROVED** [review]
Thanks for making the changes @ludwiczakpawel! 🚀

**Pull request merged**